docs/design/Network Topology Auto Discovery.md

```diff
@@ -0,0 +1,224 @@
+  topo-level-keys:
+    - network-tier1
+    - network-tier2
+    - network-tier3
+```
+This configuration instructs the controller to look for nodes with the label keys `network-tier1`，`network-tier2` and `network-tier3`
+
+**_Warning: Ensure the order of the keys is correct. Incorrect ordering of the keys will generate a topology that does not match your expectations._** 
+
+2. Assume you have k8s nodes with following labels:
+
+```yaml
+lables:
+  network-tier1: s0
+  network-tier2: s4
+  network-tier3: s6
+```
+
+```yaml
+lables:
+  network-tier1: s1
+  network-tier2: s4
+  network-tier3: s6
+```
+
+```yaml
+lables:
+  network-tier1: s2
+  network-tier2: s5
+  network-tier3: s6
+```
+
+```yaml
+lables:
+  network-tier1: s3
+  network-tier2: s5
+  network-tier3: s6
+```
+
+3. Volcano will automatically create following HyperNode resources
+`topology1-t1-s0-xxxx, topology1-t1-s1-xxxx, topology1-t1-s2-xxxx, topology1-t1-s3-xxxx, topology1-t2-s4-xxxx, topology1-t2-s5-xxxx, topology1-t3-s6-xxxx`
+
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  name: topology1-t3-s6-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: topology1-t2-s4-xxxx
+    type: HyperNode
+  - selector:
+      exactMatch:
+        name: topology1-t2-s5-xxxx
+    type: HyperNode
+  tier: 3
+```
+
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  name: topology1-t2-s4-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: topology1-t1-s0-xxxx
+    type: HyperNode
+  - selector:
+      exactMatch:
+        name: topology1-t1-s1-xxxx
+    type: HyperNode
+  tier: 2
+```
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  name: topology1-t2-s5-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: topology1-t1-s2-xxxx
+    type: HyperNode
+  - selector:
+      exactMatch:
+        name: topology1-t1-s3-xxxx
+    type: HyperNode
+  tier: 2
+```
+
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  name: topology1-t1-s0-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: node1
+    type: Node
+  - selector:
+      exactMatch:
+        name: node2
+    type: Node
+  tier: 1
+```
+
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  name: topology1-t1-s1-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: node3
+    type: Node
+  - selector:
+      exactMatch:
+        name: node4
+    type: Node
+  tier: 1
+```
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+  name: topology1-t1-s2-xxxx
+spec:
+  members:
+  - selector:
+      exactMatch:
+        name: node5
+    type: Node
+  - selector:
+      exactMatch:
+        name: node6
+    type: Node
+  tier: 1
+```
+
+```yaml
+apiVersion: topology.volcano.sh/v1alpha1
+kind: HyperNode
+metadata:
+  labels:
+    volcano.io/label-based-topology-name: topology1
```

Comment:
What does this label mean? How about adding a label key on hypedndoe to show the mapping with labels on node?